smartsim/builders/ensemble.py

```diff
@@ -197,7 +212,32 @@ def exe_arg_parameters(
         """Set the executable argument parameters.
 
         :param value: the executable argument parameters
+        :raises TypeError: if exe_arg_parameters is not mapping
+        of str and sequences of sequences of strings
         """
+        if value:
+            if not (
+                isinstance(value, collections.abc.Mapping)
+                and (
+                    all(
+                        isinstance(key, str)
+                        and isinstance(val, collections.abc.Sequence)
+                        and all(
+                            isinstance(subval, collections.abc.Sequence)
+                            for subval in val
+                        )
+                        and all(
+                            isinstance(item, str)
+                            for item in itertools.chain.from_iterable(val)
+                        )
+                        for key, val in value.items()
+                    )
+                )
+            ):
+                raise TypeError(
+                    "exe_arg_parameters argument was not of type mapping of str and sequences of sequences of strings"
+                )
```

Comment:
Style nit: This message is over 88 chars long. Consider breaking it up over multiple lines for when we turn `pylint` back on.
```suggestion
                raise TypeError(
                    "exe_arg_parameters argument was not of type mapping of str "
                    "and sequences of sequences of strings"
                )
```

Reply:
changed